armi/physics/neutronics/tests/test_crossSectionManager.py

```diff
@@ -1077,6 +1108,86 @@ def test_copyPregeneratedFiles(self):
             self.assertTrue(os.path.exists("rzmflxYA"))
 
 
+class TestCrossSectionGroupManagerWithTempGrouping(unittest.TestCase):
+    def setUp(self):
+        cs = settings.Settings()
+        cs["tempGroups"] = [300, 400, 500]
+        self.blockList = makeBlocks(11)
+        buAndTemps = (
+            (1, 340),
+            (2, 150),
+            (6, 410),
+            (10.5, 290),
+            (2.5, 360),
+            (4, 460),
+            (15, 370),
+            (16, 340),
+            (15, 700),
+            (14, 720),
+        )
+        for b, env in zip(self.blockList, buAndTemps):
+            bu, temp = env
+            comps = b.getComponents(Flags.FUEL)
+            assert len(comps) == 1
+            c = next(iter(comps))
+            c.setTemperature(temp)
+            b.p.percentBu = bu
+        core = self.blockList[0].core
+
+        def getBlocks(includeAll=True):
+            return self.blockList
+
+        core.getBlocks = getBlocks
+        for b in core.getBlocks():
+            print(b)
```

Comment:
no, if you dont do this, xsgm does all blocks in the core rather than just the block list, and you cannot do hand calcs to compare correct grouping. i would remove all blocks except those from the core, but i might have partial assemblies...